code/modules/recycling/disposal.dm

```diff
@@ -104,6 +108,14 @@
 		update()
 		return
 
+	else if(istype(I, /obj/item/storage/bag/crystal) || istype(I, /obj/item/storage/bag/ore))
+		var/obj/item/storage/bag/ore/B = I
+		to_chat(user, SPAN_NOTICE("You empty \the [B] into \the [src]."))
+		for(var/obj/item/O in B.contents)
+			B.remove_from_storage(O, src)
+		update()
+		return
```

Comment:
Shouldn't this be for every storage/bag item at this point?

Reply:
ehhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhh nah i don't think so. there are valid reasons for wanting to dump the entire bag. i could make it an intent check but that seems very clunky

Reply:
How is it clunky, just check for like harm intent and make all storage items drop all contents into the disposal unit, far cleaner and makes more sense